src/reconcile/rec_write.c

```diff
@@ -433,7 +434,8 @@ __rec_write_page_status(WT_SESSION_IMPL *session, WT_RECONCILE *r)
          * If the page state changed, the page has been written since reconciliation started and
          * remains dirty (that can't happen when evicting, the page is exclusively locked).
          */
-        if (__wt_atomic_cas32(&mod->page_state, WT_PAGE_DIRTY_FIRST, WT_PAGE_CLEAN))
+        if (atomic_compare_exchange_strong_explicit(&mod->page_state, &page_dirty_first,
+              WT_PAGE_CLEAN, memory_order_acq_rel, memory_order_acquire))
```

Comment:
Why in the failure case, relaxed is not enough?

Reply:
Perhaps I am being overly conservative. It is not clear to me that there is an advantage as the consequence is an assertion. I think this could be `relaxed` on failure, though I think it is inconsequential.